librad/src/id.rs

```diff
@@ -226,6 +397,18 @@ pub struct User {
     pub signatures: HashMap<PublicKey, EntitySignature>,
 }
 
+impl User {
+    pub fn new(name: &str, devices: impl Iterator<Item = &'static PeerId>) -> Self {
```

Comment:
```suggestion
    pub fn new(name: &str, devices: impl Iterator<Item = PeerId>) -> Self {
```

I would say the function should just own the `PeerId`s. I think it's pretty common for `new` to own the parameters it takes in.